core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -58,7 +58,7 @@ private[spark] case class CoalescedRDDPartition(
       val parentPreferredLocations = rdd.context.getPreferredLocs(rdd, p.index).map(_.host)
       preferredLocation.exists(parentPreferredLocations.contains)
     }
-    if (parents.size == 0) 0.0 else (loc.toDouble / parents.size.toDouble)
```

Comment:
There's nothing wrong with the change per se, although even if it's faster (usually this is implemented as `size == 0` inside a collection anyway), it's negligible unless it's a tight loop, and this isn't. Given that it ends up taking at least an hour of time, added up over everyone that looks at, tests and merges this, that I would prefer people not open tiny PRs that have virtually no effect.